### PR TITLE
feat : Implemented reentrancy guard

### PR DESCRIPTION
**Pull request by Akashneelesh**
- [x] issue #2 
- [ ] follows contribution [guide](../CONTRIBUTING.md)
- [ ] code change includes tests
- [ ] breaking change

Implemented reentrancy guard for the appchain contract

### CODE REVIEW

File: src/appchain.cairo [hunk 1]
```diff
@@ -15,6 +15,8 @@ mod appchain {
         OwnableComponent as ownable_cpt, OwnableComponent::InternalTrait as OwnableInternal,
         interface::IOwnable
     };
+    use openzeppelin::security::reentrancyguard::ReentrancyGuardComponent;
+    use openzeppelin::security::reentrancyguard::ReentrancyGuardComponent::InternalImpl as InternalReentrancyGuardImpl;
```
Comment (glihm) [inline src/appchain.cairo:19]: May we group both into one single use? We've tried this so far, to see how maintainable it is compared to one line for each import. Interested to have your feedback on that. :+1:

File: src/appchain.cairo [hunk 2]
```diff
@@ -74,7 +83,7 @@ mod appchain {
         fn update_state(ref self: ContractState, program_output: Span<felt252>) {
             self.config.assert_only_owner_or_operator();
 
-            // TODO(#2): reentrancy guard.
+            self.reentrancy_guard.start();
```
Comment (glihm) [inline src/appchain.cairo:86]: Shouldn't be the first statement of the function?

File: src/appchain.cairo (hunk 1 above)
Comment (Akashneelesh) [reply src/appchain.cairo]: Yes serr, This makes it much more cleaner

File: src/appchain.cairo (hunk 2 above)
Comment (Akashneelesh) [reply src/appchain.cairo:86]: Ah yes a silly mistake...

**Review by glihm — APPROVED** [review]
Thanks @Akashneelesh for this contribution! Looks good to me.

**Comment by Akashneelesh** [comment]
> Thanks @Akashneelesh for this contribution! Looks good to me.

Thank you serr

**Pull request merged**